### PR TITLE
Fix ledgersmb-admin user

### PR DESCRIPTION
**Pull request by ylavoie**
- Normalize options
- Reports all errors when possible
- Allow specifying country using name or abbreviation
- Complete documentation

### CODE REVIEW

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 1]
```diff
@@ -444,10 +450,10 @@ __END__
 
 =head1 SYNOPSIS
 
-   ledgersmb-admin user list <db-uri>
-   ledgersmb-admin user create <db-uri> [options] 
-   ledgersmb-admin user delete <db-uri>
-   ledgersmb-admin user change <db-uri>
+   ledgersmb-admin user list <db-uri> <user>
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:453]: Where does the `<user>` get parsed from the arguments? Is it stored in a variable somewhere?

Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm]: `user list` doesn't use a user tag, it lists all users. Fixed

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 2]
```diff
@@ -445,9 +451,9 @@ __END__
 =head1 SYNOPSIS
 
    ledgersmb-admin user list <db-uri>
-   ledgersmb-admin user create <db-uri> [options] 
-   ledgersmb-admin user delete <db-uri>
-   ledgersmb-admin user change <db-uri>
+   ledgersmb-admin user create <db-uri> <user> [options]
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:454]: Where is the `<user>` parsed from the command line arguments (and presumably stored in a variable)?

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 3]
```diff
@@ -475,11 +481,11 @@ User name of the created or changed user
 
 Password
 
-=item start_date
+=item start-date
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:484]: Takes a date as an argument. But which format?

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 4]
```diff
@@ -507,23 +513,23 @@ Employee manager
 
 Employee number
 
-=item is_manager
+=item is-manager
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:516]: Doesn't take an argument, but how do I know?

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 5]
```diff
@@ -112,20 +115,20 @@ sub _option_spec {
             # employee
             'dob=s' => \$dob,
             'employeenumber=s' => \$employeenumber,
-            'end_date=s' => \$end_date,
-            'is_manager!' => \$is_manager,
+            'end-date=s' => \$end_date,
+            'is-manager!' => \$is_manager,
             'manager=s' => \$manager,
             'role=s' => \$role,
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:121]: How does the user know which values can be specified for the "role" argument?

Comment (ehuelsmann) [reply lib/LedgerSMB/Admin/Command/user.pm:121]: Ok. Based on the code, I figured it out. Given that you allow individual roles to be specified, don't you think this option is useless unless we have an option to query available roles?

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 6]
```diff
@@ -112,20 +115,20 @@ sub _option_spec {
...
             'sales!' => \$is_sales,
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:122]: This is rather inconsistent: declaring a sales works with `--sales` but declaring a manager is `--is-manager` ?!

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 7]
```diff
@@ -112,20 +115,20 @@ sub _option_spec {
             # employee
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:115]: Where is `permission` applied? Or if it's applied, where is `role` applied?!

**Review by ehuelsmann — CHANGES_REQUESTED** [review]
There's more work to be done here: we need a list of available permissions if we're going to ask the names of the permissions as arguments of the `--permission` argument. Also, I've submitted a PR against this branch for further textual and consistency improvements.

File: lib/LedgerSMB/Admin/Command/user.pm (hunk 3 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm]: Fixed in documentation

File: lib/LedgerSMB/Admin/Command/user.pm (hunk 4 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm:564]: Fixed in documentation

File: lib/LedgerSMB/Admin/Command/user.pm (hunk 5 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm]: Added `user list a_user` to query current permissions,

File: lib/LedgerSMB/Admin/Command/user.pm (hunk 6 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm]: Fixed

File: cpanfile [hunk 8]
```diff
@@ -3,6 +3,7 @@
 
 requires 'perl', '5.32.0';
 
+requires 'Array::PrintCols';
```
Comment (ehuelsmann) [inline cpanfile:6]: Hmm. Why do we need this dependency? I know they are considered archaic, but Peel has formats which are useable to print table data. It's what i used to print the data about the available templates. They still work, although they lack flexibility with respect to the number of printed columns and their width.

Comment (ylavoie) [reply cpanfile:6]: We have tons of roles. Printing them vertically sorted is more usual than horizontally.

File: lib/LedgerSMB/Admin/Command/user.pm (hunk 7 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm:100]: Cleaned confusion between user role (Job title) and postgres roles (permissions)

**Review by ehuelsmann — APPROVED** [review]

**Pull request merged**